tests/test_sync.py

```diff
@@ -0,0 +1,194 @@
+        os.makedirs(venv)
+
+        with open(f"{venv}/pyvenv.cfg", "w") as f:
+            f.write("version = 3.8.16")
+
+        with open(f"{reporoot}/devenv/config.ini", "w") as f:
+            f.write(mock_config)
+
+        sync.main(context={"repo": "sentry", "reporoot": reporoot})
+        mock_volta_install.assert_called_once()
+        mock_colima_install.assert_called_once()
+        mock_limactl_install.assert_called_once()
+        mock_proc_run.assert_has_calls(
+            [
+                call(
+                    (
+                        f"{pythons_root}/{python_version}/python/bin/python3",
+                        "-m",
+                        "venv",
+                        venv,
+                    ),
+                    exit=True,
+                ),
+                call((f"{venv}/bin/sentry", "init", "--dev")),
+                call(
+                    (
+                        f"{venv}/bin/sentry",
+                        "devservices",
+                        "up",
+                        "redis",
+                        "postgres",
+                    ),
+                    exit=True,
+                ),
+            ]
+        )
+        mock_run_procs.assert_has_calls(
+            [
+                call(
+                    "sentry",
+                    reporoot,
+                    venv,
+                    (("git and precommit", ("make", "setup-git")),),
+                ),
+                call(
+                    "sentry",
+                    reporoot,
+                    venv,
+                    (
+                        ("javascript dependencies", ("make", "install-js-dev")),
+                        ("python dependencies", ("make", "install-py-dev")),
+                    ),
+                ),
+                call(
+                    "sentry",
+                    reporoot,
+                    venv,
+                    (
+                        (
+                            "python migrations",
+                            (f"{venv}/bin/sentry", "upgrade", "--noinput"),
+                        ),
+                    ),
+                ),
+            ]
+        )
+
+
+def test_linux(tmp_path: str) -> None:
+    with patch("devenv.sync.DARWIN", False), patch(
+        "devenv.lib.volta.install"
+    ) as mock_volta_install, patch(
+        "devenv.lib.colima.install"
+    ) as mock_colima_install, patch(
+        "devenv.lib.limactl.install"
+    ) as mock_limactl_install, patch(
+        "shutil.rmtree"
+    ), patch(
+        "devenv.lib.proc.run", side_effect=[None, None, None]
+    ) as mock_proc_run, patch(
+        "devenv.sync.run_procs", side_effect=[True, True, True]
+    ) as mock_run_procs:
+        reporoot = f"{tmp_path}/sentry"
+        venv = f"{tmp_path}/sentry/.venv"
+
+        os.makedirs(f"{reporoot}/devenv")
+        os.makedirs(venv)
+
+        with open(f"{venv}/pyvenv.cfg", "w") as f:
+            f.write("version = 3.8.16")
```

Comment:
Should we be hard coding this? It makes it harder to maintain

Reply:
this is fine and never needs to change because the main purpose here is to demonstrate venv recreation